Autopilot/CMakeLists.txt

```diff
@@ -272,43 +272,63 @@ elseif(${KIND_OF_BUILD} STREQUAL "SIMULATION")
   set(CMAKE_CXX_STANDARD 11)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+if( NOT EXISTS "${SIM_ROOT_DIR}/GeneratedCCode/Headers")
+  message( FATAL_ERROR "Evenin buddy. So looks like you've got an issue here. 1 of 2 things happened. First potential issue is you've forgotten to copy the unzipped GeneratedCCode directory into the Infinion folder, go and do that if you haven't. If that's done then here's your problem: see, when Simulink generates code for you it will name one of the directories inside the GeneratedCCode directory after the current matlab version. Cmake can't keep up with that as things change over the years so you're going to have to rename the R2021a (or whatever version your'e at) to be Headers. Anyway, I'll leave you alone now. Good luck with your cool simulation, I can only imagine what sort of crazy shit you're trying to get working :D." )
+endif()
 
   include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}/../Common/Inc
     ${CMAKE_CURRENT_SOURCE_DIR}/Inc/
     ${CMAKE_CURRENT_SOURCE_DIR}/AttitudeManager/Inc
+    ${CMAKE_CURRENT_SOURCE_DIR}/PathManager/Inc
+    ${CMAKE_CURRENT_SOURCE_DIR}/TelemetryManager/Inc
 
     ${SIM_ROOT_DIR}/GeneratedCCode/SimulinkModel/Infinion_grt_rtw
-    ${SIM_ROOT_DIR}/GeneratedCCode/R2020a/extern/include
-    ${SIM_ROOT_DIR}/GeneratedCCode/R2020a/rtw/c/src
-    ${SIM_ROOT_DIR}/GeneratedCCode/R2020a/rtw/c/src/ext_mode/common
-    ${SIM_ROOT_DIR}/GeneratedCCode/R2020a/simulink/include
+    ${SIM_ROOT_DIR}/GeneratedCCode/Headers/extern/include
+    ${SIM_ROOT_DIR}/GeneratedCCode/Headers/rtw/c/src
+    ${SIM_ROOT_DIR}/GeneratedCCode/Headers/rtw/c/src/ext_mode/common
+    ${SIM_ROOT_DIR}/GeneratedCCode/Headers/simulink/include
   )
```

Comment:
So from now on, we're gonna extract the generated code into this "Headers" file?

Reply:
Yeah, people who generate code will be required to rename this folder to Headers. Otherwise Cmake will break every quarter. I've got some code in the cmake that detects if this folder isn't renamed and instructs people to rename it if that's the case.